### PR TITLE
chore: backport to v0.34.x tendermint

### LINKED ISSUE #951: chore: prepare node for backport to v0.34.x tendermint
In order to be aligned with efforts made in app/core on downgrades, node should also do downgrade of respective deps 

Ref: 
- https://github.com/celestiaorg/celestia-core/pull/802
- https://github.com/celestiaorg/celestia-core/issues/790

### PR DESCRIPTION
**Pull request by distractedm1nd**
Resolves #951

### CODE REVIEW

**Comment by distractedm1nd** [comment]
@renaynay The main issue currently seems to be with how the config works in `core/testing.go` - haven't been able to figure it out yet though

**Comment by distractedm1nd** [comment]
@renaynay There are now only 4 tests that are failing, but I can't figure out why yet. 3 of them pass when run individually, but get a context timeout when run with the others for some reason. The other one, `TestFraudProofBroadcasting`, has some file write issue that I haven't been able to track down yet. Will keep looking, but if you could checkout and take a look it might be something stupid easy that I'm just overseeing

**Review by Bidon15 — COMMENTED** [review]
imho, we should wait until app/core cut proper releases to merge this awesome PR 
Thanks for the heavy-lifting @distractedm1nd 💪

File: go.mod
Comment (distractedm1nd) [reply go.mod:9]: Definitely, same with core, waiting for the release to update

Comment (evan-forbes) [reply go.mod:9]: I'll go create release candidates, one sec

File: core/testing.go
Comment (renaynay) [inline core/testing.go]: @distractedm1nd can you doc why we do this?

File: go.mod
Comment (liamsi) [reply go.mod:275]: https://github.com/celestiaorg/celestia-core/releases/tag/v1.3.1-tm-v0.34.20

**Review by liamsi — CHANGES_REQUESTED** [review]
Use the latest core release. Otherwise looks good!

File: core/testing.go
Comment (distractedm1nd) [reply core/testing.go]: Hey, think thats from an older commit, `rpctest.StopTendermint` works again :D !

File: go.mod
Comment (distractedm1nd) [reply go.mod]: done

File: go.mod
Comment (distractedm1nd) [reply go.mod]: done

**Review by liamsi — APPROVED** [review]

File: node/tests/swamp/swamp.go
Comment (renaynay) [inline node/tests/swamp/swamp.go:75]:
```suggestion
```

**Review by renaynay — COMMENTED** [review]
* downgrade rsmt2d to 0.5.0 (as core has it)
* test against backported core

File: core/testing.go
Comment (renaynay) [reply core/testing.go]: shipit

**Comment by renaynay** [comment]
We need to rebase :(

File: go.mod
Comment (Bidon15) [inline go.mod:289]:
```suggestion
	github.com/libp2p/go-libp2p-pubsub v0.7.0 => github.com/celestiaorg/go-libp2p-pubsub v0.6.2-0.20220812132010-46b2a019f2f2
```

Comment (Bidon15) [reply go.mod:289]: And go mod tidy later 👍

**Review by renaynay — APPROVED** [review]
smorge - tACKed on testground

**Review by Bidon15 — APPROVED** [review]
🚢 
worked on testground with `0.7.0-rc1` app and core 1.3.3 v0.34.20

**Review by Wondertan — APPROVED** [review]
I am actually surprised that there are such small changes in here, and I am glad that the Tendermint team decided to keep some of the good changes and reverted some weird changes like `/websocket` one and `NewBlock` type changes. Also thanks for testing this on the Testground Viet. 

Just one question that I could find the answer my self, but asking here.

**Pull request merged**
